### PR TITLE
[ntuple] Add RField description members

### PR DESCRIPTION
**Pull request by mxxo**
Allow setting RNTuple field descriptions. The immediate motivation for this PR is for use by the ongoing CMS RNTuple NanoAOD output module. The new APIs are a direct analogue to `TBranch::SetTitle`. Field descriptions are added at the `RNTupleModel` level. There currently is no way to adjust field descriptions after creation. 

There is a new  `RNTupleModel::AddField` overload that takes a description parameter and a new `RNTupleModel::MakeFieldWithDescription` method. The latter was added because of difficulties overloading the existing variadic template method `MakeField`.

### CODE REVIEW

**Review by jblomer — CHANGES_REQUESTED** [review]
Very nice!  My only concern is the name `MakeFieldWithDescription`, which is a little long for my taste.  How about `MakeField` and `AddField` overloads that take a tuple as a first parameter, so that you can write

    MakeField<float>({"pt", "transverse momentum"}, 0.0);

File: tree/ntuple/v7/test/ntuple_basics.cxx [hunk 1]
```diff
@@ -268,3 +268,64 @@ TEST(RNTupleModel, EnforceValidFieldNames)
...
+   {
+      RNTupleWriter::Recreate(std::move(model), "ntuple", fileGuard.GetPath());
+   }
+
+   auto ntuple = RNTupleReader::Open("ntuple", fileGuard.GetPath());
+   std::vector<std::string> fieldDescriptions;
+   for (auto& f: ntuple->GetDescriptor().GetTopLevelFields()) {
+      fieldDescriptions.push_back(f.GetFieldDescription());
+   }
+   ASSERT_EQ(fieldDescriptions.size(), 3);
```
Comment (jblomer) [inline tree/ntuple/v7/test/ntuple_basics.cxx:295]: Arguments of `ASSERT_EQ` and `EXPECT_EQ` should be reversed

**Comment by mxxo** [comment]
I like the tuple approach much better! I'm struggling to build master right now but will rebase with this change as soon as I can.

**Review by jalopezg-r00t — COMMENTED** [review]
I was wondering where the field description was [already] serialized/deserialized; I found it in `RNTupleDescriptor.cxx: SerializeField()`.

LGTM.  :-)

**Review by jblomer — APPROVED** [review]

**Pull request merged**